### DISCUSSION TITLE
Successful build of v1.7.0.0 on Arch Linux

### DISCUSSION THREAD

**Discussion by gonsolo** [dreamworksanimation/openmoonray#174] [General]
This is a patch to build v1.7.0.0 on Arch Linux:

[arch.patch.gz](https://github.com/user-attachments/files/18210115/arch.patch.gz)

62 files changed, 116 insertions(+), 88 deletions(-)

I am using `cmake -DMOONRAY_USE_OPTIX=NO` since I'm running on Nouveau and NVK.

**Comment by gonsolo** [comment]
Maybe the cstdint related stuff can be merged to reduce the size of the patch over time, @randypacker?

```
grep -n "+#include" arch.patch |grep cstdint
670:+#include <cstdint>
855:+#include <cstdint>
867:+#include <cstdint>
879:+#include <cstdint>
973:+#include <cstdint>
985:+#include <cstdint>
997:+#include <cstdint>
```

The rest of the patch deals with:
1. Messing up PATH in CMakeLists.txt (line 10) ([Issue](https://github.com/dreamworksanimation/openmoonray/issues/131))
2. Finding some packages (line 23)
3. Adapting to Arch's monolithic USD build (line 41 and others)
4. Commenting a fix for icc (line 60...)
5. Newer version of USD (line 183)
6. Finding python (line 382...)
7. tbb scheduler (line 435)
8. __TBB_Yield() (line 457)
9. tbb::this_task_arena (line 483)
10. std::scoped_lock instead of tbb::recursive_mutex::scoped_lock
11. Include Imath (line 658)
12. std::atomic instead of tbb::atomic (line 744...)
13. include <functional> (line 801)
14. "emit" clash (line 892)

**Comment by gonsolo** [comment]
Updated patch: [arch.patch.gz](https://github.com/user-attachments/files/19246167/arch.patch.gz)

I also created a few merge requests for the cstdint failures and point 1. above (PATH), @randypacker

**Comment by jlanz** [comment]
thanks @gonsolo - I've created an internal ticket to see if we can apply this patch.  We can't yet test on Arch linux, but we can at least test that the changes don't cause issues with Rocky9/macOS.

**Reply by gonsolo** [reply]
I'm happy to add a merge request now and then to shrink the patch. :)

**Reply by gonsolo** [reply]
I hope you are aware of the newer patch here:
https://github.com/dreamworksanimation/openmoonray/discussions/192
The biggest part of that  patch is the removal of
cmake_modules/cmake/FindTBB.cmake
which shouldn't be necessary in Rocky 9.
If you could remove that file, the rest of the patch is really small.
